sandbox/modules/identity/src/internalClusterTest/java/org/opensearch/identity/BasicAuthenticationIT.java

```diff
@@ -0,0 +1,186 @@
+        transportServices.add(clusterManagerService);
+
+        TransportService dataNodeService = internalCluster().getInstance(TransportService.class, dataNode);
+        transportServices.add(dataNodeService);
+
+        for (TransportService service : transportServices) {
+            TransportMessageListener listener = new TransportMessageListener() {
+                @Override
+                public void onRequestReceived(long requestId, String action) {
+                    final ThreadPool threadPool = internalCluster().getInstance(ThreadPool.class, service.getLocalNode().getName());
+                    Map<String, String> tcHeaders = threadPool.getThreadContext().getHeaders();
+                    if (expectedActionName.equals(action)) {
+                        if (tcHeaders.containsKey(ThreadContextConstants.OPENSEARCH_AUTHENTICATION_TOKEN_HEADER)) {
+                            interceptedTokens.put(
+                                service.getLocalNode().getId(),
+                                tcHeaders.get(ThreadContextConstants.OPENSEARCH_AUTHENTICATION_TOKEN_HEADER)
+                            );
+                        }
+                    }
+                    // String prefix = "(nodeName=" + service.getLocalNode().getId() + ", requestId=" + requestId +
+                    // ", action=" + action + ", nodeRoles=" + service.getLocalNode().getRoles() + " onRequestReceived)";
+                    // System.out.println(prefix + " Headers: " + threadPool.getThreadContext().getHeaders());
+                }
+            };
+            listenerMap.put(service.getLocalNode().getId(), listener);
+            service.addMessageListener(listener);
+        }
+
+        Request request = new Request("GET", "/_cluster/health");
+        RequestOptions options = RequestOptions.DEFAULT.toBuilder().addHeader("Authorization", "Basic YWRtaW46YWRtaW4=").build(); // admin:admin
+        request.setOptions(options);
+        List<NodeInfo> dataNodeInfos = nodeInfos.stream().filter(ni -> ni.getNode().isDataNode()).collect(Collectors.toList());
+        RestClient restClient = createRestClient(dataNodeInfos, null, "http");
+        Response response = restClient.performRequest(request);
+
+        String content = new String(response.getEntity().getContent().readAllBytes(), StandardCharsets.UTF_8);
+
+        // System.out.println("interceptedTokens: " + interceptedTokens);
+
+        assertTrue(interceptedTokens.keySet().size() == 2);
+        assertFalse(interceptedTokens.values().stream().anyMatch(s -> Objects.isNull(s)));
+
+        List<String> tokens = interceptedTokens.values().stream().collect(Collectors.toList());
+
+        boolean allEqual = tokens.isEmpty() || tokens.stream().allMatch(tokens.get(0)::equals);
+        assertTrue(allEqual);
+
+        assertEquals(200, response.getStatusLine().getStatusCode());
+        assertTrue(content.contains("\"status\":\"green\""));
```

Comment:
Seems like this failure would be hard to debug since the content isn't output, using something like `assertThat(content, containsString("green"))` would be better

Reply:
Will update, still working on a CI issue